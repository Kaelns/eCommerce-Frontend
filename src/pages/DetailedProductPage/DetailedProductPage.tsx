import { ProductHead } from '@/pages/DetailedProductPage/components/ProductHead';

// import { Stack } from '@mui/system';
// import { Paths } from '@/shared/constants';
// import type { SxStyles } from '@/shared/types';
// import CloseIcon from '@mui/icons-material/Close';
// import { useFetch } from '@/hooks/useFetch/useFetch';
// import { sxMixins } from '@/features/mui-theme/mixins';
import { ImgCarousel } from '@/features/ImgCarousel/index.ts';

// import { LoadingFetch } from '@/components/LoadingFetch';
// import { useNavigate, useParams } from 'react-router-dom';
// import { LANGUAGE, SRCSET_API } from '@/services/constants';
// import type { StackProps, SxProps, Theme } from '@mui/system';
// import { convertSxToArr } from '@/utils/convert/convertSxToArr';
// import { PageSkeleton } from '@/components/skeleton/PageSkeleton';
// import { useCallback, useContext, useMemo, useState } from 'react';
// import { findInCategories } from '@/services/ecommerce/helpers/products/findInCategories';
// import { ProductHead } from '@/pages/DetailedProductPage/components/ProductHead';
// import { getProductByKeyApi } from '@/services/model/products/getProductByKeyApi';
// import { convertToLightProduct } from '@/services/ecommerce/helpers/products/convertToLightProduct';
// import { Box, IconButton, Modal, Typography, useMediaQuery, useTheme } from '@mui/material';
import { useParams, useNavigate } from 'react-router-dom';

import { TitleText } from '@/shared/ui/elements';

import { Paths } from '@/shared/model/data';

// const sxStyles: SxStyles = {
//   modal: {
//     display: 'flex',
//     alignItems: 'center',
//     justifyContent: 'center'
//   },
//   modalBody: {
//     position: 'relative',
//     width: '93%',
//     height: { zero: '54%', tablet: '80%', laptop: '93%' },
//     padding: 1,
//     borderRadius: 1,
//     bgcolor: 'common.background'
//   },
//   imgContainer: {
//     position: 'relative',
//     borderRadius: 1,
//     cursor: 'pointer',
//     overflow: 'hidden',
//     ...sxMixins.mediaHover(
//       {
//         transform: 'scale(1.05)'
//       },
//       '> img'
//     )
//   },
//   closeIcon: {
//     position: 'absolute',
//     width: 50,
//     height: 50,
//     top: 0,
//     right: 0,
//     zIndex: 100,
//     transform: { zero: 'translate(25%, -25%)', tablet: 'translate(50%, -50%)' },
//     bgcolor: 'primary.main',
//     ...sxMixins.mediaHover({
//       bgcolor: 'primary.light'
//     })
//   }
// };

export function DetailedProductPage() {
  const { id: key } = useParams();
  const navigate = useNavigate();

  if (!key) {
    navigate(Paths.ERROR);
  }

  // const { data, isLoading, error } = useFetch(getProductByKeyApi, key!);
  // const { categories } = useContext(ECommerceContext);
  // const productData = useMemo(() => convertToLightProduct(data), [data]);
  // // TODO swap categories on categoriesObj
  // const categoriesNames = useMemo(
  //   () => findInCategories(categories, productData.categoriesIdArr, true).map((obj) => obj.name[LANGUAGE]),
  //   [categories, productData.categoriesIdArr]
  // );

  // const theme = useTheme();
  // const isMatchesMedia = useMediaQuery(theme.breakpoints.down('tablet'));

  // const [open, setOpen] = useState(false);
  // const [imgNum, setImgNum] = useState(0);
  // const handleClose = (): void => setOpen(false);
  // const handleOpen = useCallback(
  //   (num: number) => (): void => {
  //     setImgNum(num);
  //     setOpen(true);
  //   },
  //   []
  // );

  // const createImages = useCallback(
  //   <T extends StackProps['height']>(
  //     { imgHeight, imgStyles = {}, containerStyles = {} }: ICreateImagesStyles<T>,
  //     onClick?: (num: number) => () => void
  //   ) => {
  //     const maxSize = imgHeight && 'maxSize' in imgHeight ? imgHeight.maxSize : undefined;
  //     return productData.images.map((imageData, index) => (
  //       <ImgLoad
  //         key={imageData.url}
  //         src={imageData.url}
  //         alt={productData.name}
  //         height={imgHeight?.height}
  //         sx={imgStyles}
  //         containerStyles={[sxStyles.imgContainer, ...convertSxToArr(containerStyles)]}
  //         srcset={{ srcSetArr: SRCSET_API, maxSize }}
  //         onClick={onClick ? onClick(index) : undefined}
  //       />
  //     ));
  //   },
  //   [productData]
  // );

  return (
    <TitleText>Something</TitleText>
    // <LoadingFetch error={error} isLoading={isLoading} Skeleton={PageSkeleton}>
    //   <Stack gap={1.5} flexDirection={{ zero: 'column-reverse', tablet: 'column' }}>
    //     <Stack direction="row" justifyContent="space-between" gap={1.5}>
    //       <ImgCarousel
    //         width={{ zero: 1, tablet: '65%' }}
    //         customDots={createImages({ imgHeight: { height: 100 }, containerStyles: { width: 130 } })}
    //       >
    //         {createImages({ imgHeight: { height: 300 } }, handleOpen)}
    //       </ImgCarousel>
    //       {!isMatchesMedia && <ProductHead productData={productData} categoriesNames={categoriesNames} />}
    //     </Stack>

    //     <Box>
    //       {isMatchesMedia && <ProductHead productData={productData} categoriesNames={categoriesNames} />}
    //       <Typography>
    //         <b> &emsp;Description: </b>
    //         {productData.description}
    //       </Typography>
    //     </Box>

    //     <Modal open={open} onClose={handleClose} sx={sxStyles.modal}>
    //       <Stack direction="row" alignItems="center" justifyContent="center" sx={sxStyles.modalBody}>
    //         <IconButton onClick={handleClose} sx={sxStyles.closeIcon}>
    //           <CloseIcon />
    //         </IconButton>
    //         <ImgCarousel arrows openModalImg={imgNum}>
    //           {createImages({
    //             imgHeight: { height: { zero: '40vh', tablet: '65vh', laptop: '85vh' }, maxSize: 'unlimited' }
    //           })}
    //         </ImgCarousel>
    //       </Stack>
    //     </Modal>
    //   </Stack>
    // </LoadingFetch>
  );
}
