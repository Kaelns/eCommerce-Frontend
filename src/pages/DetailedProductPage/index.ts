export { DetailedProductPage } from '@/pages/DetailedProductPage/DetailedProductPage';
export { prefetchDetailedProductPage } from '@/pages/DetailedProductPage/model/prefetchDetailedProductPage.loader';
