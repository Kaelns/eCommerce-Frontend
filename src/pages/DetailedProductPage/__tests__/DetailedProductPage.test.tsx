import { DetailedProductPage } from '@/pages/DetailedProductPage/DetailedProductPage';
import { render } from '@testing-library/react';

describe('Given DetailedProductPage component: ', () => {
  it('when rendered, should match snapshot', () => {
    const { asFragment } = render(<DetailedProductPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
