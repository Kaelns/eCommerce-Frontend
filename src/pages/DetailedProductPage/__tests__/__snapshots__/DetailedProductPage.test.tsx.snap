// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given DetailedProductPage component:  > when rendered, should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-79elbk"
  >
    <div
      class="MuiStack-root css-r78jfo-MuiStack-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-7bduwa-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1lbmg9-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-13v5yvq-MuiSkeleton-root"
      />
    </div>
    <div
      class="MuiStack-root css-x6bnna-MuiStack-root"
    >
      <div
        class="MuiStack-root css-1shucjb-MuiStack-root"
      >
        <div
          class="MuiStack-root css-re6u3m-MuiStack-root"
        >
          <div
            class="slick-slider MuiBox-root css-zz3cru slick-initialized"
            dir="ltr"
          >
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              />
            </div>
          </div>
          <div
            class="MuiStack-root css-9f25q2-MuiStack-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1yej70y-MuiTypography-root"
            />
            <div
              class="MuiStack-root css-1hgk1tk-MuiStack-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-iy6nlt-MuiTypography-root"
              >
                Price: 
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-11y6jfv-MuiTypography-root"
              >
                0 $
              </h6>
            </div>
            <div
              class="MuiStack-root css-wjj0nm-MuiStack-root"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-13hdmgp-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-1415yxo-MuiLinearProgress-root"
                role="progressbar"
              >
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-eg0r9f-MuiLinearProgress-bar1"
                />
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1yf7653-MuiLinearProgress-bar2"
                />
              </span>
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-57pk4j-MuiTypography-root"
              >
                Add to
              </h6>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-p9fvvg-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2m10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2m-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
          >
            <b>
               â€ƒDescription: 
            </b>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
